internal/services/datadog/datadog_monitors_resource.go

```diff
@@ -0,0 +1,493 @@
+							Sensitive: true,
+						},
+
+						"application_key": {
+							Type:      pluginsdk.TypeString,
+							Required:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"enterprise_app_id": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"linking_auth_code": {
+							Type:      pluginsdk.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"linking_client_id": {
+							Type:      pluginsdk.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"redirect_uri": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"identity": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"SystemAssigned",
+								"UserAssigned",
+							}, false),
+						},
+
+						"principal_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"tenant_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"sku": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"user_info": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogUsersName,
+						},
+
+						"email_address": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogMonitorsEmailAddress,
+						},
+
+						"phone_number": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DatadogMonitorsPhoneNumber,
+						},
+					},
+				},
+			},
+
+			"monitoring_status": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"liftr_resource_category": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"liftr_resource_preference": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"marketplace_subscription_status": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"type": {
```

Comment:
We don't need to expose `type` here, and above computed properties if they won't be used in another resource, we don't need to expose them either.

Reply:
Oh okay, so we can remove these parameters?

Reply:
Yes

Reply:
We need these properties as user might need the values.